types/recurly.ts

```diff
@@ -0,0 +1,34 @@
+import Risk from './3d-secure';
+import Adyen from './adyen';
+import ApplePay from './apple-pay';
+import BankAccount from './bank-account';
+import Configure from './configure';
+import Elements from './elements';
+import Emitter from './emitter';
+import GiftCard from './gift-card';
+import PayPal from './paypal';
+import Pricing from './pricing';
+import Token from './token';
+import Validate from './validate';
+
+type RecurlyEvent = 'change' | 'field:submit' | 'error';
+
+interface Recurly extends Emitter<RecurlyEvent> {
+    Adyen: Adyen;
+    ApplePay: ApplePay;
+    bankAccount: BankAccount;
+    configure: Configure;
+    Elements: Elements;
+    giftCard: GiftCard;
+    PayPal: PayPal;
+    Pricing: Pricing;
+    Risk: Risk;
+    token: Token;
+    validate: Validate;
+}
+
+declare global {
+    interface Window {
+        recurly: Recurly;
+    }
+}
```

Comment:
sorry to nitpick: some declarations have terminating semicolons and others do not. Should we lint types?

Reply:
We definitely should, great suggestion.

The semi-colon stuff is pretty interesting. Apparently interfaces shouldn't have semi-colons at the end of their bodies, but types should https://github.com/typescript-eslint/typescript-eslint/issues/1444

There's no eslint rule to enforce semi-colons at the end of function bodies either.

I actually think the best route to enforce some consistency with semi-colon usage would be adding https://eslint.org/docs/2.0.0/rules/no-extra-semi

Let's have a conversation about it and figure out a plan 👍